src/vector_index.h

```diff
@@ -0,0 +1,86 @@
+#include "search_ctx.h"
+#include "VecSim/vecsim.h"
+#include "index_iterator.h"
+
+#define VECSIM_TYPE_FLOAT32 "FLOAT32"
+#define VECSIM_TYPE_FLOAT64 "FLOAT64"
+#define VECSIM_TYPE_INT32 "INT32"
+#define VECSIM_TYPE_INT64 "INT64"
+
+#define VECSIM_METRIC_IP "IP"
+#define VECSIM_METRIC_L2 "L2"
+
+#define VECSIM_ALGORITHM_BF "BF"
+#define VECSIM_ALGORITHM_HNSW "HNSW"
+
+#define VECSIM_INITIAL_CAP "INITIAL_CAP"
+#define VECSIM_M "M"
+#define VECSIM_EF "EF"
+
+typedef enum {
+  VECTOR_TYPE_FLOAT32 = 0,
+  VECTOR_TYPE_DOUBLE64 = 1,
+  VECTOR_TYPE_INT32 = 2,
+  VECTOR_TYPE_INT64 = 3,
+  VECTOR_TYPE_BYTE = 4,
+  VECTOR_TYPE_BOOL = 5,
+} VectorType;
+
+typedef enum {
+  VECTOR_ALG_BF = 0,
+  VECTOR_ALG_HNSW = 1,
+} VectorAlgorithm;
+
+
+typedef enum {
+  VECTOR_TOPK = 0,
+  VECTOR_RANGE = 1,
+} VectorQueryType;
+
+typedef enum {
+  VECTOR_DIST_L2 = 0,
+  VECTOR_DIST_COSINE = 1,
+  VECTOR_DIST_IP = 2,
+} VectorDistanceMetric;
```

Comment:
These guys are defined in `VecSim/vecsim.h`. Do we need to re-define them here?

Reply:
removed